### PR TITLE
Add Go v1.22.x to CI workflow

### PR DESCRIPTION
**Pull request by VirrageS**
This adds `1.22.x` Go version to CI workflow so we also test against this version to make sure everything works fine.

### CODE REVIEW

**Comment by klauspost** [comment]
I think it is ok to remove 1.18 support. It is far beyond EOL.

File: .github/workflows/validate.yml [hunk 1]
```diff
@@ -21,7 +21,7 @@ jobs:
   linters:
     strategy:
       matrix:
-        go-version: [1.21.x]
+        go-version: [1.21.x, 1.22.x]
```
Comment (klauspost) [inline .github/workflows/validate.yml:24]:
```suggestion
        go-version: [1.22.x]
```
Just latest should be fine for this.

**Comment by VirrageS** [comment]
@klauspost It is your call, I can keep v1.18 for now.

**Comment by klauspost** [comment]
Remove it in the PR. Having 4 versions back should be more than enough.

**Comment by VirrageS** [comment]
From what I see the https://github.com/philhofer/fwd/commit/fbbf4953d98612dac320a9051da34e3e761d1df7 requires at least 1.20. And from what I see 1.22 build requires that we use this new `unsafe.Slice`. So I think we would need to drop 1.19 too :(

**Comment by klauspost** [comment]
I'm not going to object to that. Personally I support last 3 versions, which is already one more than Go, but I don't know if Phil has an opinion on this.

I suggest you add that to this PR, we can adjust if needed.

**Review by klauspost — APPROVED** [review]
LGTM

**Comment by philhofer** [comment]
Dropping support for EOL'd go versions is fine to me. LGTM.

**Pull request merged**